src/WebJobs.Script.WebHost/FileMonitoringService.cs

```diff
@@ -24,7 +24,7 @@ public class FileMonitoringService : IHostedService, IDisposable
     {
         private readonly ScriptJobHostOptions _scriptOptions;
         private readonly IScriptEventManager _eventManager;
-        private readonly IApplicationLifetime _applicationLifetime;
+        private readonly AspNetCore.Hosting.IApplicationLifetime _applicationLifetime;
```

Comment:
Lots of namespace updates here -- I've checked and we don't use `Microsoft.Extensions.Hosting.IApplicationLifetime` anywhere anymore.

Reply:
It appears that this interface is now obsolete https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.hosting.iapplicationlifetime?view=aspnetcore-3.1. They say we should be using Microsoft.Extensions.Hosting.IHostApplicationLifetime now?

Reply:
That's new for 3.0 -- but for now the old interface is supported as well. They have their `ApplicationLifetime` class implement both the old and the new: https://github.com/dotnet/aspnetcore/blob/master/src/Hosting/Hosting/src/Internal/ApplicationLifetime.cs#L15

To take a dependency on the new interface would require some changes in 3.0. They'd be good and make this less subtle -- we'd have to explicitly make sure we're calling `StopApplication()` on the WebHost so it'd be clearer what we're doing. But I think we should handle that separately...

Reply:
Added a comment at the PR level. We should stick to `IApplicationLifetime` for now for the reasons outlined there.